### PR TITLE
Port exit this page from 4.7

### PR DESCRIPTION
**Pull request by owenatgov**
Ports the [exit this page component](https://design-system.service.gov.uk/components/exit-this-page/) from the 4.7.0 release to `main`, including refactoring the js to ES2015 in line with https://github.com/alphagov/govuk-frontend/pull/3771

Includes changes from the following PRs:
- https://github.com/alphagov/govuk-frontend/pull/2545
- https://github.com/alphagov/govuk-frontend/pull/3874
- https://github.com/alphagov/govuk-frontend/pull/3878

Additionally supersedes https://github.com/alphagov/govuk-frontend/pull/3880

Fixes https://github.com/alphagov/govuk-frontend/issues/3840

### CODE REVIEW

File: packages/govuk-frontend/src/govuk/components/exit-this-page/exit-this-page.mjs [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+    this.keypressTimeoutId = null
+
+    /** @private */
+    this.timeoutMessageId = null
+  }
+
+  /**
+   * Initialise component
+   */
+  init () {
```
Comment (colinrotherham) [inline packages/govuk-frontend/src/govuk/components/exit-this-page/exit-this-page.mjs:98]: Thanks for moving this higher up 😊

File: packages/govuk-frontend/src/govuk/components/exit-this-page/accessibility.test.mjs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+describe('/components/exit-this-page', () => {
+  describe('component examples', () => {
+    let exampleNames
+
+    beforeAll(async () => {
+      exampleNames = Object.keys(await getExamples('exit-this-page'))
+    })
+
+    it('passes accessibility tests', async () => {
```
Comment (colinrotherham) [inline packages/govuk-frontend/src/govuk/components/exit-this-page/accessibility.test.mjs:12]: Fab work. Cheers for moving the accessibility tests into the new format 🙌

**Review by colinrotherham — CHANGES_REQUESTED** [review]
Nicely done @owenatgov

Few minor snags but nothing big at all. I think we'd best split out non-port changes (like renaming selector properties) unless it was for consistency with v5

File: packages/govuk-frontend/src/govuk/components/exit-this-page/exit-this-page.test.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      ])
+
+      const url = new URL(await page.url())
+      expect(url.pathname).toBe(pathname)
+    })
+
+    it('navigates to the href of the skiplink', async () => {
+      await goToExample(page, 'exit-this-page-with-skiplink')
+
+      const href = await page.$eval(buttonClass, el => el.getAttribute('href'))
```
[the remark this answers is not in the document]
Comment (colinrotherham) [reply packages/govuk-frontend/src/govuk/components/exit-this-page/exit-this-page.test.js:34]: Sorry @owenatgov if we're keeping `buttonClass` can we update the `it()` description?

>navigates to the href of the ~skiplink~ button

[the remark this answers is not in the document]
Comment (colinrotherham) [reply packages/govuk-frontend/src/govuk/components/exit-this-page/exit-this-page.test.js:34]: There's one on line 66 to check too

[the remark this answers is not in the document]
Comment (owenatgov) [reply packages/govuk-frontend/src/govuk/components/exit-this-page/exit-this-page.test.js:34]: Ah sorry I misread your comment! Will update now.

**Review by colinrotherham — APPROVED** [review]
🚀

**Pull request merged**